Validation/SiTrackerPhase2V/plugins/Phase2OTValidateTTStub.cc

```diff
@@ -81,48 +128,279 @@ Phase2OTValidateTTStub::~Phase2OTValidateTTStub() {
 }
 
 void Phase2OTValidateTTStub::dqmBeginRun(const edm::Run &iRun, const edm::EventSetup &iSetup) {
-  tkGeom_ = &(iSetup.getData(geomToken_));
-  tTopo_ = &(iSetup.getData(topoToken_));
+  theTrackerGeom_ = &(iSetup.getData(geomToken_));
+  tTopo_ = &(iSetup.getData(tTopoToken_));
+
+  // Clear existing histograms
```

Comment:
Where does the idea to clear the existing histograms come from? It may be correct, but I don't see simiilar things happening in beginRun of the other DQM packages. e.g. https://github.com/brandiskip/cmssw/blob/test_branch/Validation/SiTrackerPhase2V/plugins/Phase2OTValidateCluster.cc .

Reply:
I kept getting a seg fault and someone pointed me to lines 116 and 117 found here: https://cmssdt.cern.ch/lxr/source/DQM/HLTEvF/plugins/TriggerBxVsOrbitMonitor.cc?v=CMSSW_14_1_X_2024-07-17-2300
and when I followed that format, it fixed my issues. Perhaps there is a better way. I will investigate further.

Reply:
You need to set the vector size to 6 or whatever, since you refer to the elements later on. e.g. https://github.com/brandiskip/cmssw/blob/test_branch/Validation/SiTrackerPhase2V/plugins/Phase2OTValidateTTStub.cc#L452 . But I don't understand why it needs resetting o 6 every run. Or why the call to clear is needed. If it were set to 6 in the constructor, resize would not be needed.